src/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -541,32 +524,67 @@ private static void OnAppContextUnloading(object sender, EventArgs e)
         private static readonly object s_initLock = new object();
 
         // Occurs when an Assembly is loaded
-        public static event AssemblyLoadEventHandler AssemblyLoad
+        public static event AssemblyLoadEventHandler AssemblyLoad;
+
+        // Occurs when resolution of type fails
+        public static event ResolveEventHandler TypeResolve;
+
+        // Occurs when resolution of resource fails
+        public static event ResolveEventHandler ResourceResolve;
+
+        // Occurs when resolution of assembly fails
+        // This event is fired after resolve events of AssemblyLoadContext fails
+        public static event ResolveEventHandler AssemblyResolve;
+
+        // This method is called by the VM.
+        private static void OnAssemblyLoad(RuntimeAssembly assembly)
         {
-            add { AppDomain.CurrentDomain.AssemblyLoad += value; }
-            remove { AppDomain.CurrentDomain.AssemblyLoad -= value; }
+            AssemblyLoad?.Invoke(null /* AppDomain */, new AssemblyLoadEventArgs(assembly));
```

Comment:
These events got the internal AppDomain type as a sender before this change (that code outside CoreLib cannot do much with). They are getting null as a sender now. Nobody reported it as a problem so far.

Reply:
Since it was an internal type, presumably if it were to become an issue we could just use a dummy singleton `new object()`?

Reply:
It is the actual public AppDomain type in .NET Framework. I think if somebody hits compat issue with this, it is most likely going to be dependency on the public AppDomain type. We would need to callback from CoreFX to notify about the public AppDomain type to fix that, or something similar.